### PR TITLE
Enabling feature loadbalancer

### PR DESCRIPTION
**Pull request by vishalanarase**

### CODE REVIEW

**Comment by alejandrojnm** [comment]
@vishalanarase this ready to merge ??

**Comment by vishalanarase** [comment]
@alejandrojnm No, It just for review. We can merge, But this code will not work until we merge the API endpoint.

File: loadbalancer_test.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+		     "public_ip": "10.90.100.37",
+		     "private_ip": "192.168.1.2",
+		     "firewall_id": "d8e8bae3-5a76-435e-b96d-24955deb8792",
+		     "state": "available"
+		  }
+		]`,
+	})
+	defer server.Close()
+	got, err := client.ListLoadBalancers()
+
```
Comment (RealHarshThakur) [inline loadbalancer_test.go:39]:
```suggestion
```

File: loadbalancer.go [hunk 2]
```diff
@@ -9,51 +9,47 @@ import (
...
-	Port       int    `json:"port"`
+	IP              string `json:"ip"`
+	Protocol        string `json:"protocol,omitempty"`
+	SourcePort      int32  `json:"source_port"`
+	TargetPort      int32  `json:"target_port"`
+	HealthCheckPort int32  `json:"health_check_port,omitempty"`
 }
 
 // LoadBalancer represents a load balancer configuration within Civo
 type LoadBalancer struct {
```
Comment (RealHarshThakur) [inline loadbalancer.go:29]: Should we have separate struct for "Create" , etc? We've seen this causing confusion when request and response params are in the same struct.

Comment (vishalanarase) [reply loadbalancer.go:29]: `LoadBalancer` is the response struct and `LoadBalancerConfig` is used to create. We follow the same pattern for all objects.

Comment (RealHarshThakur) [reply loadbalancer.go:29]: How do we differentiate between Create and Update request structs for `LoadBalancerConfig`?

File: kubernetes.go [hunk 3]
```diff
@@ -300,3 +300,52 @@ func (c *Client) ListAvailableKubernetesVersions() ([]KubernetesVersion, error)
...
 }
+
+// ListKubernetesClusterInstances returns all cluster instances
+func (c *Client) ListKubernetesClusterInstances(id string) ([]Instance, error) {
+	resp, err := c.SendGetRequest(fmt.Sprintf("/v2/kubernetes/clusters/%s/instances", id))
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	instances := []Instance{}
```
Comment (RealHarshThakur) [inline kubernetes.go:311]:
```suggestion
	instances := make([]Instance,0)
```

File: loadbalancer.go (hunk 2 above)
Comment (vishalanarase) [reply loadbalancer.go:29]: Create new struct for update request `LoadBalancerUpdateConfig`

File: kubernetes.go (hunk 3 above)
Comment (vishalanarase) [reply kubernetes.go:311]: Done

File: loadbalancer.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+// LoadBalancer represents a load balancer configuration within Civo
+type LoadBalancer struct {
+	ID                           string                `json:"id"`
+	Name                         string                `json:"name"`
+	Algorithm                    string                `json:"algorithm"`
+	Backends                     []LoadBalancerBackend `json:"backends"`
+	ExternalTrafficPolicy        string                `json:"external_traffic_policy,omitempty"`
+	SessionAffinity              string                `json:"session_affinity,omitempty"`
+	SessionAffinityConfigTimeout int32                 `json:"session_affinity_config_timeout,omitempty"`
+	EnableProxyProtocol          bool                  `json:"enable_proxy_protocol,omitempty"`
```
Comment (RealHarshThakur) [inline loadbalancer.go:37]: Some description/comment about what this does?

**Pull request merged**